[Alerting][Docs] Fix rule types categorization (#118285)
elastic/kibana: docs/user/alerting/rule-types.asciidoc

[[rule-types]]
== Rule types

> A rule is a set of <<alerting-concepts-conditions, conditions>>, <<alerting-concepts-scheduling, schedules>>, and <<alerting-concepts-actions, actions>> that enable notifications. {kib} provides rules built into the Elastic Stack and rules registered by one of the {kib} apps.

Reviewer: Instead of talking about the UI in each section, I think it would be better to add this paragraph after the intro paragraph:

You can create most rules types in [Stack Management > Rules and Connectors](https://www.elastic.co/guide/en/kibana/current/create-and-manage-rules.html). For information on creating security rules, refer to [Create a detection rule](https://www.elastic.co/guide/en/security/current/rules-ui-create.html).

[[stack-rules]]
=== Stack rules

> <<create-and-manage-rules, Stack rules>> are built into {kib}. To access the *Stack Rules* feature and create and edit rules, users require the `all` privilege. See <<kibana-feature-privileges, feature privileges>> for more information. In the full rules list in the UI, Stack rules are shown under the proper category `STACK RULES`.

Reviewer: If you add the intro above, you can remove the line "In the full rules list ..."

[cols="2*<"]
|===

| <<rule-type-es-query>>
| Run a user-configured {es} query, compare the number of matches to a configured threshold, and schedule actions to run when the threshold condition is met.

| <<rule-type-index-threshold>>
| Aggregate field values from documents using {es} queries, compare them to threshold values, and schedule actions to run when the thresholds are met.

| {ref}/transform-alerts.html[{transform-cap} rules] beta:[]
| beta:[] Run scheduled checks on a {ctransform} to check its health. If a {ctransform} meets the conditions, an alert is created and the associated action is triggered.

> | <<geo-alerting, Tracking containment>>

Reviewer: The beta tag appears twice in Transform Rules.  I suggest removing the tag under the title.

| Run an {es} query to determine if any documents are currently contained in any boundaries from a specified boundary index and generate alerts when a rule's conditions are met.

|===

[float]
[[observability-rules]]
=== Observability rules

> In the rules list in the Management UI, Observability rules are shown under the multiple categories `APM AND USER EXPERIENCE`, `LOGS`, `METRICS`, `STACK MONITORING` and `UPTIME`.

Reviewer: Maybe something a little more generic:

Observability rules categorized into APM and User Expereince, Logs, Metrics, Stack Monitoring, and Uptime.

[cols="2*<"]
|===

| <<apm-alerts, APM and User Experience>>

> | Allows to define rules to detect complex conditions within *APM* data and trigger built-in actions when those conditions are met.

Reviewer: Make this text consistent with the rest.

```suggestion
| Detect complex conditions in *APM* data and trigger built-in actions when the conditions are met.
```

| {observability-guide}/create-alerts.html[Logs rules]
| Detect complex conditions in the *Logs* app.

| {observability-guide}/create-alerts.html[Metrics rules]
| Detect complex conditions in the *Metrics* app.

| <<kibana-alerts,Stack Monitoring>>
| Provide {kib} Alerting rules out-of-the box to notify you of potential issues in the Elastic Stack.

| {observability-guide}/create-alerts.html[Uptime rules]
| Detect complex conditions in the *Uptime* app.

|===

[float]
[[ml-rules]]
=== Machine learning rules

> beta:[] {ml-docs}/ml-configuring-alerts.html[{ml-cap} rules] run scheduled checks on an {anomaly-job} to detect anomalies with certain conditions. If an anomaly meets the conditions, an alert is created and the associated action is triggered. In the rules list in the Management UI, Machine learning rules are shown under the proper category `MACHINE LEARNING`.

Reviewer: remove the last sentence?

[[stack-rules]]
=== Stack rules

<<create-and-manage-rules, Stack rules>> are built into {kib}. To access the *Stack Rules* feature and create and edit rules, users require the `all` privilege. See <<kibana-feature-privileges, feature privileges>> for more information.

[cols="2*<"]
|===

| <<rule-type-es-query>>
| Run a user-configured {es} query, compare the number of matches to a configured threshold, and schedule actions to run when the threshold condition is met.

| <<rule-type-index-threshold>>
| Aggregate field values from documents using {es} queries, compare them to threshold values, and schedule actions to run when the thresholds are met.

| {ref}/transform-alerts.html[{transform-cap} rules]
| beta:[] Run scheduled checks on a {ctransform} to check its health. If a {ctransform} meets the conditions, an alert is created and the associated action is triggered.

| <<geo-alerting, Tracking containment>>
| Run an {es} query to determine if any documents are currently contained in any boundaries from a specified boundary index and generate alerts when a rule's conditions are met.

|===

[float]
[[observability-rules]]
=== Observability rules

> Observability rules categorized into APM and User Expereince, Logs, Metrics, Stack Monitoring, and Uptime.

Reviewer 2: ```suggestion
Observability rules are categorized into APM and User Experience, Logs, Metrics, Stack Monitoring, and Uptime.
```

> [cols="2*<"]

Reviewer 2: There is a section above that says `{kib} provides two types of rules: rules specific to the Elastic Stack and rules specific to a domain.`. Should that be reworded as well? Should we remove the references to "two types of rules"? Maybe `{kib} provides rules built into the Elastic Stack and rules registered by one of the {kib} apps.`
